### PR TITLE
Add a git_repo role

### PR DESCRIPTION
**Pull request by wbclark**
This role uses a single ansible data structure to manage a set of git repositories with potentially multiple remotes and multiple branches per remote.

### CODE REVIEW

**Comment by wbclark** [comment]
Not yet properly tested, due to difficulty getting #1579 to play nicely with https://github.com/theforeman/forklift/commit/5683c94fb95fca963f98512d5094c4b7e964c52d

**Comment by wbclark** [comment]
I added a README for this role explaining how it is used, and marked it ready for review

**Comment by ianballou** [comment]
I haven't tested this entire thing, but it has been in use with our hotfixing process for over a year now and has been working well.

File: roles/git_repo/tasks/branch.yml [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+- name: "Create local repository branch tracking specified remote branch"
+  ansible.builtin.command:
```
Comment (wbclark) [inline roles/git_repo/tasks/branch.yml:3]: I've tried to use the git_config module for this sort of thing, which was suggested by the maintainers of the git module as that module doesn't support everything that we need to do here, but branch creation was the one area where I could never git_config to work.

Doing it with the command module smells somewhat hacky, but it's not only worked very reliably, it also ends up being quite a bit simpler.

**Review by ianballou — COMMENTED** [review]
This is looking good to me! How I tested:

```
[ian@manicotto forklift]$ cat playbooks/repositories_test.yml
---
- hosts: all
  tasks:
    - name: "Setup git repositories"
      ansible.builtin.include_role:
        name: git_repo
      loop: "{{ git_repositories }}"
      loop_control:
        loop_var: repo
      vars:
        git_repo_repository: "{{ repo }}"
```

```
  fedora39:
    box_name: 'fedora/39-cloud-base'
    pty: true
    ansible:
      playbook: 'playbooks/repositories_test.yml'
      variables:
        git_repositories:
          - name: 'foreman'
            dir: '/home/vagrant'
            remotes:
              - name: 'origin'
                url: 'https://github.com/theforeman/foreman.git'
                branches:
                  - 'develop'
          - name: 'katello'
            dir: '/home/vagrant'
            remotes:
              - name: 'origin'
                url: 'https://github.com/Katello/katello.git'
                branches:
                  - 'master'
            python_packages:
              - 'ansible'
              - 'obal'
              - 'tito'
 ```

Only snag I hit was that git wasn't on my box. Should a role be responsible for ensuring certain dependencies are installed? I'm not sure if that's a usual Ansible expectation.

**Comment by ianballou** [comment]
(For future reference, I took that playbook from here: https://github.com/wbclark/forklift/commit/bb6de0d3b7fe24d11730045a2b27c3202ce8221b)

**Review by ianballou — APPROVED** [review]
Looks good to me! In the future we can revamp the git role to use the generic package action so that other distros are supported.

**Review by evgeni — APPROVED** [review]
As said on Matrix: I think it's too complicated, but I don't mind having it if it's useful for y'all

Mergin.

**Pull request merged**